hazelcast/src/main/java/com/hazelcast/map/impl/operation/MapReplicationStateHolder.java

```diff
@@ -271,7 +271,7 @@ public void readData(ObjectDataInput in) throws IOException {
         for (int i = 0; i < size; i++) {
             String name = in.readUTF();
             int numOfRecords = in.readInt();
-            List keyRecord = new ArrayList<>(numOfRecords * 2);
+            List keyRecord = new LinkedList();
```

Comment:
What is the reasoning to replace the sized `ArrayList`s with `LinkedList`s here and in other places? I guess adding each key & record in a `LinkedList` will introduce a penalty (wrapper `LinkedList$Node` objects being created & memory access pattern being irregular) vs the preallocated array in `ArrayList`.

Reply:
seems a lefover, reverted this change.